2020-06/transpose/transpose.py

```diff
@@ -1,2 +1,44 @@
 def transpose(lines):
+
+    n_text=len(lines) #VGG number of letters in the full input set
+
+    in_data=lines.split("\n")
+    n_lines=len(in_data) #VGG number of text lines
+    
+    n=max([len(text) for text in in_data])
+    out=["" for i in range(n)]
+    output=""
+    
+    for i in range(n):
+        for text in in_data:
+            if len(text)>i:
+                out[i]+=text[i:i+1]
+            else:
+
+                #VGG fixing the test_second_line_longer_than_first_line        
+                if text==in_data[0]:
+                    out[i]+=" "
+
+                #VGG fixing the test_triangle
+                if len(out[i])>1 and len(out[i].strip())==0:
+                    out[i]+=" "
+
+                #VGG fixing the test_mixed_line_length
+                if len(out[i]) ==1 and (
+                    text!=in_data[0] and out[i][0:1] !="."
+                    and out[i-1][-1:-2:-1] !="." and len(out[i-1])!=1 ):
+                        out[i]+=" "
```

Comment:
This was a test I struggled to pass as well. I believe this is fixing the left-padding in the case where we get a long line, then a short line then a long line again, but I'm struggling to interpret this conditional statement :flushed: (it's pretty complicated).

I'm going to merge this in shortly, but if you get a second can you shed some light on how this is doing what it's doing?